crates/router/src/macros.rs

```diff
@@ -51,3 +51,18 @@ macro_rules! async_spawn {
         tokio::spawn(async move { $t });
     };
 }
+
+#[macro_export]
+macro_rules! get_keys_of_option_nones {
+    [$(($key:expr, $option:expr)), *] => {
```

Comment:
```suggestion
    [$(($key:literal, $option:expr)),+] => {
```
- replace `*` with `+` since calling the macro with 0 tuples makes no sense.
- `$key:literal`: we can start with the assumption that we will always statically know the names of fields when invoking the macro. Can be updated to something more permissive like `expr` later if required.